tests/kubernetes/minikube.py

```diff
@@ -0,0 +1,204 @@
+        self.kubeconfig = "/tmp/scratch/kubeconfig-%s" % self.container.id[:12]
+        time.sleep(2)
+        rc, output = self.container.exec_run("cp -f %s %s" % (kubeconfig_path, self.kubeconfig))
+        assert rc == 0, "failed to get %s from minikube!\n%s" % (kubeconfig_path, output.decode('utf-8'))
+        time.sleep(2)
+        kube_config.load_kube_config(config_file=self.kubeconfig)
+
+    def connect(self, name, timeout, version=None):
+        print("\nConnecting to %s container ..." % name)
+        self.container = self.host_client.containers.get(name)
+        self.client = self.get_client()
+        self.name = name
+        self.version = version
+        self.load_kubeconfig(timeout=timeout)
+
+    def deploy(self, version, timeout, name=None, options={}):
+        self.version = version
+        if self.version[0] != 'v':
+            self.version = 'v' + self.version
+        self.name = name
+        if not self.name:
+            self.name = "minikube-%s-%s" % (MINIKUBE_VERSION, self.version)
+        if not options:
+            options = {
+                "name": self.name,
+                "privileged": True,
+                "environment": {
+                    'K8S_VERSION': self.version,
+                    'TIMEOUT': str(timeout)
+                },
+                "ports": {
+                    '8080/tcp': None,
+                    '8443/tcp': None,
+                    '2375/tcp': None,
+                },
+                "volumes": {
+                    "/tmp/scratch": {
+                        "bind": "/tmp/scratch",
+                        "mode": "rw"
+                    },
+                }
+            }
+        print("\nDeploying minikube %s cluster ..." % self.version)
+        image, logs = self.host_client.images.build(
+            path=os.path.join(TEST_SERVICES_DIR, 'minikube'),
+            buildargs={"MINIKUBE_VERSION": MINIKUBE_VERSION},
+            tag="minikube:%s" % MINIKUBE_VERSION,
+            rm=True,
+            forcerm=True)
+        self.container = self.host_client.containers.run(
+            image.id,
+            detach=True,
+            **options)
+        self.load_kubeconfig(timeout=timeout)
+        self.container.reload()
+        self.get_client()
+        self.get_ip()
+
+    def create_secret(self, key, secret):
+        if self.container:
+            rc, output = self.container.exec_run("kubectl create secret generic %s --from-literal=access-token=%s" % (key, secret))
+            assert rc == 0
+            print_lines(output.decode('utf-8'))
+
+    @contextmanager
+    def deploy_yamls(self, yamls=[], services_dir=K8S_SERVICES_DIR):
+        self.yamls= []
+        if services_dir:
+            if len(yamls) == 0:
+                yamls = sorted([os.path.join(services_dir, y) for y in os.listdir(services_dir) if y.endswith(".yaml")])
```

Comment:
It is better to have a separate function called `deploy_service_yamls` or something that does the `os.listdir` and calls `deploy_yamls` with those files, that way the difference in encoded in the function name and this function is simpler.

Reply:
I'll remove this for now, and re-implement if necessary based on how the monitor tests are restructured.